redash/models.py

```diff
@@ -757,6 +759,94 @@ def __unicode__(self):
         return unicode(self.id)
 
 
+class AccessPermission(BaseModel):
+
+    id = peewee.PrimaryKeyField()
+    object_type = peewee.CharField()
+    object_id = peewee.IntegerField()
+    object = GFKField('object_type', 'object_id')
+    access_type = peewee.CharField()
+    grantor = peewee.ForeignKeyField(User, related_name='grantor')
+    grantee = peewee.ForeignKeyField(User, related_name='grantee')
```

Comment:
How about we make `grantee` a GFK field as well? So we can later on add the option to grant to permission to an `ApiKey` object or a `Group` object.

It's NTH so we can skip this for now, as it should be an easy migration.